src/xenia/kernel/xam/xam_ui.cc

```diff
@@ -87,30 +87,24 @@ class MessageBoxDialog : public xe::ui::ImGuiDialog {
 };
 
 // http://www.se7ensins.com/forums/threads/working-xshowmessageboxui.844116/?jdfwkey=sb0vm
-SHIM_CALL XamShowMessageBoxUI_shim(PPCContext* ppc_context,
-                                   KernelState* kernel_state) {
-  uint32_t user_index = SHIM_GET_ARG_32(0);
-  uint32_t title_ptr = SHIM_GET_ARG_32(1);
-  uint32_t text_ptr = SHIM_GET_ARG_32(2);
-  uint32_t button_count = SHIM_GET_ARG_32(3);
-  uint32_t button_ptrs = SHIM_GET_ARG_32(4);
-  uint32_t active_button = SHIM_GET_ARG_32(5);
-  uint32_t flags = SHIM_GET_ARG_32(6);
-  uint32_t result_ptr = SHIM_GET_ARG_32(7);
-  uint32_t overlapped_ptr = SHIM_GET_ARG_32(8);
-
+dword_result_t XamShowMessageBoxUI(dword_t user_index, lpstring_t title_ptr,
+                                   lpstring_t text_ptr, dword_t button_count,
```

Comment:
Both `title_ptr` and `text_ptr` could be just `title` and `text`. Please refer to other usages of `lpstring_t`/`lpwstring_t`.